server/lib/utils/util.go

```diff
@@ -50,3 +53,48 @@ func IFIP(name string, need_ipv6 bool) (net.IP, error) {
 	}
 	return nil, fmt.Errorf("interface %v: no suitable setting found", name)
 }
+
+type WaitGroup struct {
+	sync.WaitGroup
+	errors []error
+	cnt int32
+}
+
+func NewWaitGroup(size int) *WaitGroup {
+	wg := &WaitGroup {
+		errors: make([]error, size),
+	}
+	wg.WaitGroup.Add(size)
+	return wg
+}
+
+func (wg *WaitGroup) Done(err error) {
```

Comment:
__[gometalinter]__
* [golint] exported method WaitGroup.Done should have comment or be unexported